### PR TITLE
Add potentialDupesIgnoringUsername. Fixes #2537

### LINKED ISSUE #2537: Function that returns list of duplicate logins ignoring username.
E.g. the usernames might disagree, but we should ignore that. Request from lockwise meeting.

CC @jhugman

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-749)

### PR DESCRIPTION
**Pull request by thomcc**
~~Needs tests, getting this up now though.~~

Should be good to go once green.

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/logins/src/db.rs [hunk 1]
```diff
@@ -856,6 +856,44 @@ impl LoginDb {
...
 
+    pub fn potential_dupes_ignoring_username(&self, login: &Login) -> Result<Vec<Login>> {
+        // Could be lazy_static-ed...
+        lazy_static::lazy_static! {
+            static ref DUPES_IGNORING_USERNAME_SQL: String = format!(
+                "SELECT {common_cols} FROM loginsL
+                WHERE is_deleted = 0
+                    AND hostname = :hostname
+                    AND (
+                        formSubmitURL = :form_submit
```
Comment (grigoryk) [inline components/logins/src/db.rs]: In https://github.com/mozilla/application-services/blame/master/components/logins/src/db.rs#L404 you're doing something subtly different, why is that?

**Comment by ekager** [comment]
Okay I attempted to set up AS, rebase this, and add a test that seems to be passing (?!). But it might be absolute gibberish! 🤷‍♀️ Feel free to trash if it's way off the mark / we can chat in the next few days abt it 
https://github.com/mozilla/application-services/compare/master...ekager:potential-tests

File: components/logins/src/db.rs (hunk 1 above)
Comment (thomcc) [reply components/logins/src/db.rs]: 1. The way we dedupe during syncs is necessarially different. `find_dupe` is only used during sync. I think there's a patch somewhere that renames this to make it clearer but there are... many dead patches around duping. in logins (see 2)
2. The story around logins dupes is... Storied. TLDR that code is wrong, but it's tricky to fix. We have semi-regular meetings talking about this and other duping-related issues.

You'll note that this is in-line with how we detect dupes https://github.com/mozilla/application-services/blob/master/components/logins/src/db.rs#L819.

**Comment by thomcc** [comment]
> Okay I attempted to set up AS, rebase this, and add a test that seems to be passing (?!). But it might be absolute gibberish!

That looks perfect, not gibberish at all.

I'll put that in the patch and get it reviewed/landed if y'all would be happy with this as a solution for #2878.

**Comment by thomcc** [comment]
Oh, I didn't notice last night that you also moved these to use protobufs. I... Wish I had noticed that before doing the same.

File: components/logins/ios/Logins/LoginsStorage.swift [hunk 2]
```diff
@@ -372,6 +372,20 @@ open class LoginsStorage {
         }
     }
 
+    /// Get the set of potential duplicates ignoring the username of `login`.
+    open func potentialDupesIgnoringUsername(to login: LoginRecord) throws -> [LoginRecord] {
+        let json = try login.toJSON()
+        return try queue.sync {
+            let engine = try self.getUnlocked()
+            let buffer = try LoginsStoreError.unwrap { err in
+                sync15_passwords_potential_dupes_ignoring_username(engine, json, err)
```
Comment (eoger) [inline components/logins/ios/Logins/LoginsStorage.swift]: This needs to be updated to protobuf input also?

Comment (thomcc) [reply components/logins/ios/Logins/LoginsStorage.swift]: Forgot to save before committing 😅

**Comment by thomcc** [comment]
(Note about the diabled lint rule: It's specifically complaining that there are more than 200 lines in `LoginStorage` type declaration. This doesn't seem like something we care about anywhere else, so I think it's fine to disable)

**Pull request merged**